src/template_finder.py

```diff
@@ -222,6 +249,27 @@ def search_all(
             break
     return matches
 
+def search_all_templates(
```

Comment:
Why can you not use search_all() here?

Reply:
If we really need the list[Template] as input, we should add it to the serach_all() function.

Reply:
This suggests shape of img and roi are incompatible. 

But rather than going down that rabbit hole we should just refactor search_all(). Alex pointed out there's a better way with cv2. https://discord.com/channels/904057828791185458/906298605420285993/978388103800713316

Reply:
To clarify for @Ezro and @aeon0 

`search_all()` is for finding all matches from a list of templates on screen. It returns a list of match objects. It iterates over the cv2.matchTemplate result and continuously masks it until there's no results left. So I input ["quarter", "dime"] and the picture were a picture of a ton of coins, it'd result a list of all the quarter and dime matches in the image. 

`search_all()` works, but it's not optimal. the optimal method is as `_alex_` described in that discord link. 

`search_all_templates()` returns the first match of all templates of interest on the screen. So if I input ["quarter", "dime"] and the picture were a picture of a ton of coins, it'd result the first match of quarter and the first match of dime. 

Perhaps there's a way we can better name these functions? Or restructure template_finder somehow to accommodate?

Reply:
But esentially the two functions do more or less very similar things and also share the same code structure. Couldnt we do it with a flag?

Reply:
Resolving this as the PR has been updated to use a flag in search_all